src/disco/FixedWidthField/Real/test/parseExponent.test.cpp

```diff
@@ -1,24 +1,103 @@
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "E+123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "E+ 123", position ) );
+    CHECK( 6 == position );
+    CHECK( +123 == parse( "E123", position ) );
+    CHECK( 4 == position );
+    CHECK( +123 == parse( "E 123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "e+123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "e+ 123", position ) );
+    CHECK( 6 == position );
+    CHECK( +123 == parse( "e123", position ) );
+    CHECK( 4 == position );
+    CHECK( +123 == parse( "e 123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "D+123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "D+ 123", position ) );
+    CHECK( 6 == position );
+    CHECK( +123 == parse( "D123", position ) );
+    CHECK( 4 == position );
+    CHECK( +123 == parse( "D 123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "d+123", position ) );
+    CHECK( 5 == position );
+    CHECK( +123 == parse( "d+ 123", position ) );
+    CHECK( 6 == position );
+    CHECK( +123 == parse( "d123", position ) );
+    CHECK( 4 == position );
+    CHECK( +123 == parse( "d 123", position ) );
+    CHECK( 5 == position );
+
+    CHECK( -123 == parse( "-123", position ) );
+    CHECK( 4 == position );
+    CHECK( -123 == parse( "- 123", position ) );
+    CHECK( 5 == position );
+    CHECK( -123 == parse( "E-123", position ) );
+    CHECK( 5 == position );
+    CHECK( -123 == parse( "E- 123", position ) );
+    CHECK( 6 == position );
+    CHECK( -123 == parse( "e-123", position ) );
+    CHECK( 5 == position );
+    CHECK( -123 == parse( "e- 123", position ) );
+    CHECK( 6 == position );
+    CHECK( -123 == parse( "D-123", position ) );
+    CHECK( 5 == position );
+    CHECK( -123 == parse( "D- 123", position ) );
+    CHECK( 6 == position );
+    CHECK( -123 == parse( "d-123", position ) );
+    CHECK( 5 == position );
+    CHECK( -123 == parse( "d- 123", position ) );
+    CHECK( 6 == position );
+
+    // exponent is only read up to the first non-digit
+    CHECK( +123 == parse( "+123 ", position ) );
+    CHECK( 4 == position );
+    CHECK( +123 == parse( "+123a", position ) );
+    CHECK( 4 == position );
+  }
+
+  THEN( "invalid exponent strings cause an exception" )
+  {
+    uint16_t position = 0;
+
+    CHECK_THROWS( parse( "-a123", position ) );
+    CHECK_THROWS( parse( "+a123", position ) );
+    CHECK_THROWS( parse( "E-a123", position ) );
+    CHECK_THROWS( parse( "E+a123", position ) );
+    CHECK_THROWS( parse( "D-a123", position ) );
+    CHECK_THROWS( parse( "D+a123", position ) );
+    CHECK_THROWS( parse( "E -123", position ) );
+    CHECK_THROWS( parse( "E +123", position ) );
+    CHECK_THROWS( parse( "D -123", position ) );
+    CHECK_THROWS( parse( "D +123", position ) );
+
+    CHECK_THROWS( parse( "-  a3", position ) );
+    CHECK_THROWS( parse( "-    ", position ) );
```

Comment:
Fortunately, this one is not valid.  I did confirm that because I was concerned.